db/migrate/20191023180130_update_openstack_hosts_uid_ems.rb

```diff
@@ -0,0 +1,23 @@
+class UpdateOpenstackHostsUidEms < ActiveRecord::Migration[5.1]
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+
+  def up
+    Host.in_my_region.where(:type => "ManageIQ::Providers::Openstack::InfraManager::Host").find_each do |host|
+      next if host.ems_ref_obj.nil?
```

Comment:
You can probably push this into the query... `.where.not(:ems_ref_obj => nil)`

Reply:
:+1:

Reply:
Done [here](https://github.com/ManageIQ/manageiq-schema/pull/426/files#diff-13bdd75d8c9d067286df49a626dc45e1R8) and test added